service/history/mutableStateBuilder.go

```diff
@@ -1378,6 +1606,8 @@ func (e *mutableStateBuilder) AddDecisionTaskScheduledEventAsHeartbeat(
 		return nil, e.createInternalServerError(opTag)
 	}
 
+	isFirstDecision := !e.HasProcessedOrPendingDecisionTask()
```

Comment:
Can this move into the if !bypassTaskGeneration block

Reply:
this attr is going to be removed in my next PR

btw, `!e.HasProcessedOrPendingDecisionTask()` has to be called before any mutation is done to mutable state